app/controllers/desks_controller.rb

```diff
@@ -38,4 +41,8 @@ def load_desk
   def desk_params
     params.require(:desk).permit(:name, :occupied)
   end
+
+  def broadcast_needed?
+    !@desk.previous_changes.slice('name', 'occupied').empty?
+  end
```

Comment:
The response logic: determines whether a broadcast is necessary given changes.

Reply:
Beautiful. I'd avoid the double negative and do `@desk.previous_changes.slice('name', 'occupied').any?`